### PR TITLE
Juxhen applicant info

### PR DESCRIPTION
**Pull request by Juxhen**

### CODE REVIEW

File: s3_project/classes/applicant_info_class.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class ApplicantInfoClean(ExtractFromS3):
+
+    def __init__(self):
+        super().__init__()
+        self.iterate()
+
+    # This method iterates through each file, and applies the cleaning methods to each file.
+    # This method also appends the cleaned files to a dictionary.
+    def iterate(self):
```
Comment (davidrichardharvey) [inline s3_project/classes/applicant_info_class.py]: This is a bad method name

File: s3_project/classes/applicant_info_class.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        object_dict = {}
+        talent_json_dict = {}
+        for file in self.talent_json_list:
+            s3_object = self.s3_client.get_object(Bucket=self.bucket_name, Key=file)
+            body = s3_object['Body'].read()
+            object_dict = json.loads(body)
```
Comment (davidrichardharvey) [inline s3_project/classes/applicant_info_class.py:22]: I thought the extraction is defined already in the ExtractFromS3 - why is it re-coded here?

File: s3_project/classes/applicant_info_class.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    # This method changes result, self_dev, financial_support and geo_flex to boolean values.
+    def boolean_values(self, object_dict):
+        list_of_booleans = [object_dict['result'], object_dict['self_development'],
+                            object_dict['financial_support_self'], object_dict['geo_flex']]
+        list_of_booleans = list(map(lambda x: True if x == 'Yes' or x == 'Pass' else False, list_of_booleans))
+        object_dict['result'] = list_of_booleans[0]
+        object_dict['self_development'] = list_of_booleans[1]
+        object_dict['financial_support_self'] = list_of_booleans[2]
+        object_dict['geo_flex'] = list_of_booleans[3]
```
Comment (davidrichardharvey) [inline s3_project/classes/applicant_info_class.py]: Would it not be better to define a function that applies this transformation, that you can apply to each list?  Seems odd to bring them together just to split them up again.

File: s3_project/classes/applicant_info_class.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        return [object_dict['result'], object_dict['self_development'], object_dict['financial_support_self'], object_dict['geo_flex']]
+
+juxhen = ApplicantInfoClean()
```
Comment (davidrichardharvey) [inline s3_project/classes/applicant_info_class.py:83]: This line should not be in the code you push to the repo.

File: s3_project/tests/test_applicant_info_class.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+
+# Tests that the names are split up correctly
+def test_split_names():
+    assert testing.split_names(dict_1) == ['Shaun', 'Blake']
+    assert testing.split_names(dict_2) == ['Vincent van', 'Goph']
+    assert testing.split_names(dict_3) == ['Harry James Daniel', 'Peter']
+
+
+test_split_names()
```
Comment (davidrichardharvey) [inline s3_project/tests/test_applicant_info_class.py:17]: Are we not using pytest or unittest?  Why not?

File: s3_project/tests/test_applicant_info_class.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+dict_2 = {'name': 'Vincent van Goph', 'date':'23//01/2020', 'result': 'Fail', 'self_development': 'Yes', 'financial_support_self': 'No', 'geo_flex':'Yes'}
+dict_3 = {'name': 'Harry James Daniel Peter', 'date':'12//08//2012', 'result': 'Pass', 'self_development': 'Yes', 'financial_support_self': 'Yes', 'geo_flex':'Yes'}
+
+testing = ApplicantInfoClean()
+
+
+# Tests that the names are split up correctly
+def test_split_names():
+    assert testing.split_names(dict_1) == ['Shaun', 'Blake']
+    assert testing.split_names(dict_2) == ['Vincent van', 'Goph']
```
Comment (davidrichardharvey) [inline s3_project/tests/test_applicant_info_class.py:13]: But 'Van Goph' is the surname...?

**Review by davidrichardharvey — CHANGES_REQUESTED** [review]
This isn't quite ready to merge - check my comments, please.

File: s3_project/classes/applicant_info_class.py (hunk 1 above)
Comment (Juxhen) [reply s3_project/classes/applicant_info_class.py]: Changed

File: s3_project/classes/applicant_info_class.py (hunk 2 above)
Comment (Juxhen) [reply s3_project/classes/applicant_info_class.py:22]: The extraction class extracts the filename as a key, in this method we access the body of that key

File: s3_project/classes/applicant_info_class.py (hunk 3 above)
Comment (Juxhen) [reply s3_project/classes/applicant_info_class.py]: can you explain that a bit more to me when you have time?

File: s3_project/classes/applicant_info_class.py (hunk 4 above)
Comment (Juxhen) [reply s3_project/classes/applicant_info_class.py:83]: I have removed that line now

File: s3_project/tests/test_applicant_info_class.py (hunk 5 above)
Comment (Juxhen) [reply s3_project/tests/test_applicant_info_class.py:17]: Pytest was not working for Alex so he used the function to test it.

File: s3_project/tests/test_applicant_info_class.py (hunk 6 above)
Comment (Juxhen) [reply s3_project/tests/test_applicant_info_class.py:13]: we were told to put everything in the first name and flag it for now which is what was done.

File: s3_project/classes/applicant_info_class.py (hunk 3 above)
Comment (davidrichardharvey) [reply s3_project/classes/applicant_info_class.py]: You join together 'result', 'self_development', 'financial_support_self', and 'geo_flex' into a list, only to have to reassign them individually again.  Wouldn't it make more sense to write the transformation as a function that you can then apply to each item?

File: s3_project/classes/applicant_info_class.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        return [object_dict['result'], object_dict['self_development'], object_dict['financial_support_self'],
+                object_dict['geo_flex']]
+
+    def create_dataframe(self, object_dict):
+        data = object_dict
+        df = pd.DataFrame.from_dict(data)
+        print(df)
+
+
+juxhen = ApplicantInfoClean()
```
Comment (davidrichardharvey) [inline s3_project/classes/applicant_info_class.py]: Nope!

**Review by davidrichardharvey — CHANGES_REQUESTED** [review]

**Pull request closed without merging**